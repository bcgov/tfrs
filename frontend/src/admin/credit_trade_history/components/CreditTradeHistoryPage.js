import React from 'react';

import CreditTradeHistoryTable from './CreditTradeHistoryTable';

const CreditTradeHistoryPage = props => (
  <div className="page_user_activity">
    <h1>User Activity</h1>
    <CreditTradeHistoryTable />
  </div>
);

export default CreditTradeHistoryPage;
