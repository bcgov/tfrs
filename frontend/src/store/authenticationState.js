import { loadUser } from 'redux-oidc';
import { put, takeLatest, all } from 'redux-saga/effects';
import userManager from './oidc-usermanager';
import { getLoggedInUser } from '../actions/userActions';
import CONFIG from '../config';

const LOGIN_TRIGGERING_ACTIONS = [
  'redux-oidc/USER_EXPIRED'
];

function triggerLoginFlow (store) {
  const { routing } = store.getState();

  if (routing.location &&
    routing.location.pathname !== '/authCallback') {
    return CONFIG.KEYCLOAK.CUSTOM_LOGIN ? userManager.signinSilent() : userManager.signinRedirect();
  }

  return false;
}

function * getBackendUser () {
  yield put(getLoggedInUser());
}

export default function * authenticationStateSaga (store) {
  userManager.clearStaleState();

  const { routing } = store.getState();

  if (!routing.location || routing.location.pathname !== '/authCallback') {
    loadUser(store, userManager);
  }
  yield all([
    takeLatest('redux-oidc/USER_FOUND', getBackendUser),
    takeLatest(
      action => (LOGIN_TRIGGERING_ACTIONS.includes(action.type)),
      triggerLoginFlow,
      store
    )
  ]);
}
