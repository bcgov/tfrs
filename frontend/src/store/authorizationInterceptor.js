import axios from 'axios';
import userManager from './oidc-usermanager';
import CONFIG from '../config';

function configureAxios () {
  if (CONFIG.KEYCLOAK) {
    console.log('adding token to request');

    axios.interceptors.request.use((config) => {
      const loadconfig = async () => {
        const user = await userManager.getUser();
        return {
          ...config,
          headers: {
            Authorization: `Bearer ${user.id_token}`
          }
        };
      };

      return loadconfig();
    });
  }
}

export default configureAxios;
