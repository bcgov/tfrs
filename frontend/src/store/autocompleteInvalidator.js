import { call, put, takeLatest } from 'redux-saga/effects';
import AutocompleteActionTypes from "../constants/actionTypes/Autocomplete";
import FuelCodeActionTypes from '../constants/actionTypes/FuelCodes';

const INVALIDATING_ACTIONS = [
  FuelCodeActionTypes.ADD_FUEL_CODE,
  FuelCodeActionTypes.RECEIVE_FUEL_CODE,
  FuelCodeActionTypes.RECEIVE_FUEL_CODES,
  FuelCodeActionTypes.UPDATE_FUEL_CODE,
  FuelCodeActionTypes.DELETE_FUEL_CODE
];

function * invalidateCache () {
  yield put({ type: AutocompleteActionTypes.INVALIDATE_AUTOCOMPLETE_CACHE });
}

export default function * sessionTimeoutSaga () {
  yield takeLatest(
    action => (INVALIDATING_ACTIONS.includes(action.type)),
    invalidateCache
  );
}
