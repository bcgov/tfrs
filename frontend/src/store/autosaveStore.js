import {takeEvery, all, call} from 'redux-saga/effects';
import {delay} from "redux-saga";
import axios from "axios";
import * as Routes from "../constants/routes";

function* loadAutosaveData(action) {
  const {key, resolve, reject} = action.payload;

  axios
    .get(`${Routes.BASE_URL}${Routes.AUTOSAVE_API}?key=${key}`)
    .then((response) => {
      try {
        const parsed = JSON.parse(response.data.data);
        resolve(parsed);
      } catch (e) {
        reject(e);
      }
    }).catch((error) => {
    reject(error);
  });
}

function* saveAutosaveData(action) {
  const {key, state, resolve, reject} = action.payload;

  const payload = {
    key,
    data: JSON.stringify(state)
  };

  axios
    .post(`${Routes.BASE_URL}${Routes.AUTOSAVE_API}`, payload)
    .then((response) => {
      resolve();
    }).catch((error) => {
    reject(error);
  });
}

function* clearAutosaveData(action) {
  const {resolve, reject} = action.payload;

  axios
    .post(`${Routes.BASE_URL}${Routes.AUTOSAVE_API}/clear`)
    .then((response) => {
      try {
        resolve();
      } catch (e) {
        reject(e);
      }
    }).catch((error) => {
    reject(error);
  });
}

export default function* autosaveSaga() {

  yield all([
    takeEvery('LOAD_AUTOSAVE_DATA', loadAutosaveData),
    takeEvery('SAVE_AUTOSAVE_DATA', saveAutosaveData),
    takeEvery('CLEAR_AUTOSAVE_DATA', clearAutosaveData)
  ]);
}
