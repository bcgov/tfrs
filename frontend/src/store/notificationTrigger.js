import {call, put, takeLatest} from 'redux-saga/effects';
import {delay} from 'redux-saga';
import {getNotifications, getNotificationsCount} from "../actions/notificationActions";
import UserActionTypes from "../constants/actionTypes/Users";
import NotificationActionTypes from "../constants/actionTypes/Notifications";


const TRIGGERING_ACTIONS = [
  'SERVER_INITIATED_NOTIFICATION_RELOAD',
  UserActionTypes.RECEIVE_LOGGED_IN_USER,
  NotificationActionTypes.SUCCESS_NOTIFICATIONS
];

function* fetchNotifications(store) {
  yield call(delay, 1000); //debounce

  if (store.getState().rootReducer.userRequest.isAuthenticated) {
    yield put(getNotificationsCount());
  }

  if (store.getState().rootReducer.notifications.onNotificationsPage) {
    yield put(getNotifications());
  }

}

export default function* notificationsSaga(store) {
  yield takeLatest(
    action => (TRIGGERING_ACTIONS.includes(action.type)),
    fetchNotifications,
    store
  );
}
