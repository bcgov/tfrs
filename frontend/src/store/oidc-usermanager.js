import { createUserManager } from 'redux-oidc';
import CONFIG from '../config';

const settings = {
  authority: CONFIG.KEYCLOAK.AUTHORITY,
  client_id: CONFIG.KEYCLOAK.CLIENT_ID,
  redirect_uri: CONFIG.KEYCLOAK.CALLBACK_URL,
  automaticSilentRenew: true,
  silent_redirect_uri: `${window.location.protocol}//${window.location.hostname}${window.location.port ? `:${window.location.port}` : ''}/token_renew.html`,
  response_type: 'token id_token'
};

const userManager = createUserManager(settings);

export default userManager;
