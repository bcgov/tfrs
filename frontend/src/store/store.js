import { createStore, compose, applyMiddleware, combineReducers } from 'redux';
import { reducer as toastrReducer } from 'react-redux-toastr';
import { routerReducer, routerMiddleware } from 'react-router-redux';
import { createLogger } from 'redux-logger';
import io from 'socket.io-client';
import thunk from 'redux-thunk';
import rootReducer from '../reducers/reducer';
import { getNotifications } from '../actions/notificationActions';
import { SOCKETIO_URL } from '../constants/routes';

import createOidcMiddleware, { loadUser, processSilentRenew, reducer as OIDCReducer } from 'redux-oidc';
import { persistTargetPathReducer } from '../reducers/persistTargetPathReducer';

import userManager from './oidc-usermanager';
import CONFIG from '../config';
import { getLoggedInUser } from '../actions/userActions';

import persistState from 'redux-localstorage';
import createSocketIoMiddleware from 'redux-socket.io';

const middleware = routerMiddleware(history);
const oidcMiddleware = createOidcMiddleware(userManager);

const socket = io(SOCKETIO_URL);
const socketIoMiddleware = createSocketIoMiddleware(socket, 'socketio/');

const enhancer = compose(persistState(['targetPath'], {key: 'tfrs-state'}));

const combinedReducers = (state = {}, action) => {
  const currentRoute = state.routing || {};
  return {
    toastr: toastrReducer(state.toastr, action),
    oidc: OIDCReducer(state.oidc, action),
    routing: routerReducer(state.routing, action),
    targetPath: persistTargetPathReducer(state.targetPath, {...action, currentRoute}),
    rootReducer: rootReducer(state.rootReducer, action),
  };
};

let allMiddleware = [
  thunk,
  socketIoMiddleware,
  oidcMiddleware,
  middleware
];

if (process.env.NODE_ENV !== 'production') {
  allMiddleware.push(createLogger());
}

const store = createStore(
  combinedReducers,
  applyMiddleware(...allMiddleware),
  enhancer
);

let subscriptionProcessing = false;

store.subscribe(() => {
  const state = store.getState();
  if (!subscriptionProcessing) {
    subscriptionProcessing = true;

    if (state.rootReducer.notifications.serverInitiatedReloadRequested === true) {
      store.dispatch(getNotifications());
    }

    if (CONFIG.KEYCLOAK.ENABLED) {
      if (state.oidc.user &&
        !state.rootReducer.userRequest.isFetching &&
        !state.rootReducer.userRequest.isAuthenticated) {
        store.dispatch(getLoggedInUser());
      }
    }

    subscriptionProcessing = false;
  }
});

if (CONFIG.KEYCLOAK.ENABLED) {
  loadUser(store, userManager);
  processSilentRenew();
}

export default store;
