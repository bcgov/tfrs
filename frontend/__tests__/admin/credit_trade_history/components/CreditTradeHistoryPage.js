import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import CreditTradeHistoryPage from '../../../../src/admin/credit_trade_history/components/CreditTradeHistoryPage';
import store from '../../../../src/store/store';

test('CreditTradeHistoryPage should display', () => {
  const component = renderer.create(
    <Provider store={store}>
      <CreditTradeHistoryPage />
    </Provider>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
