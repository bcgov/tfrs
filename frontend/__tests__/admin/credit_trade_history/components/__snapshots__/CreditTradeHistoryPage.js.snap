// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditTradeHistoryPage should display 1`] = `
<div
  className="page_user_activity"
>
  <h1>
    User Activity
  </h1>
  <div
    className="ReactTable"
    style={Object {}}
  >
    <div
      className="rt-table"
      role="grid"
    >
      <div
        className="rt-thead -header"
        style={
          Object {
            "minWidth": "475px",
          }
        }
      >
        <div
          className="rt-tr"
          role="row"
        >
          <div
            className="rt-th rt-resizable-header -cursor-pointer"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "50 0 auto",
                "maxWidth": null,
                "width": "50px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              User
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
          <div
            className="rt-th rt-resizable-header -cursor-pointer"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "50 0 auto",
                "maxWidth": null,
                "width": "50px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              Action Taken
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
          <div
            className="rt-th -cursor-pointer"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": "100px",
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div
              className=""
            >
              Transaction ID
            </div>
          </div>
          <div
            className="rt-th rt-resizable-header -cursor-pointer"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              Initiator
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
          <div
            className="rt-th rt-resizable-header -cursor-pointer"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              Respondent
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
          <div
            className="rt-th rt-resizable-header -sort-desc -cursor-pointer"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "75 0 auto",
                "maxWidth": null,
                "width": "75px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              Timestamp
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
        </div>
      </div>
      <div
        className="rt-tbody"
        style={
          Object {
            "minWidth": "475px",
          }
        }
      >
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          >
            <div
              className="rt-td col-user"
              role="gridcell"
              style={
                Object {
                  "flex": "50 0 auto",
                  "maxWidth": null,
                  "width": "50px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-action"
              role="gridcell"
              style={
                Object {
                  "flex": "50 0 auto",
                  "maxWidth": null,
                  "width": "50px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-id"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": "100px",
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-initiator"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-respondent"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-timestamp"
              role="gridcell"
              style={
                Object {
                  "flex": "75 0 auto",
                  "maxWidth": null,
                  "width": "75px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          >
            <div
              className="rt-td col-user"
              role="gridcell"
              style={
                Object {
                  "flex": "50 0 auto",
                  "maxWidth": null,
                  "width": "50px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-action"
              role="gridcell"
              style={
                Object {
                  "flex": "50 0 auto",
                  "maxWidth": null,
                  "width": "50px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-id"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": "100px",
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-initiator"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-respondent"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-timestamp"
              role="gridcell"
              style={
                Object {
                  "flex": "75 0 auto",
                  "maxWidth": null,
                  "width": "75px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          >
            <div
              className="rt-td col-user"
              role="gridcell"
              style={
                Object {
                  "flex": "50 0 auto",
                  "maxWidth": null,
                  "width": "50px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-action"
              role="gridcell"
              style={
                Object {
                  "flex": "50 0 auto",
                  "maxWidth": null,
                  "width": "50px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-id"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": "100px",
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-initiator"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-respondent"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-timestamp"
              role="gridcell"
              style={
                Object {
                  "flex": "75 0 auto",
                  "maxWidth": null,
                  "width": "75px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          >
            <div
              className="rt-td col-user"
              role="gridcell"
              style={
                Object {
                  "flex": "50 0 auto",
                  "maxWidth": null,
                  "width": "50px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-action"
              role="gridcell"
              style={
                Object {
                  "flex": "50 0 auto",
                  "maxWidth": null,
                  "width": "50px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-id"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": "100px",
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-initiator"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-respondent"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-timestamp"
              role="gridcell"
              style={
                Object {
                  "flex": "75 0 auto",
                  "maxWidth": null,
                  "width": "75px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          >
            <div
              className="rt-td col-user"
              role="gridcell"
              style={
                Object {
                  "flex": "50 0 auto",
                  "maxWidth": null,
                  "width": "50px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-action"
              role="gridcell"
              style={
                Object {
                  "flex": "50 0 auto",
                  "maxWidth": null,
                  "width": "50px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-id"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": "100px",
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-initiator"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-respondent"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-timestamp"
              role="gridcell"
              style={
                Object {
                  "flex": "75 0 auto",
                  "maxWidth": null,
                  "width": "75px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          >
            <div
              className="rt-td col-user"
              role="gridcell"
              style={
                Object {
                  "flex": "50 0 auto",
                  "maxWidth": null,
                  "width": "50px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-action"
              role="gridcell"
              style={
                Object {
                  "flex": "50 0 auto",
                  "maxWidth": null,
                  "width": "50px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-id"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": "100px",
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-initiator"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-respondent"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-timestamp"
              role="gridcell"
              style={
                Object {
                  "flex": "75 0 auto",
                  "maxWidth": null,
                  "width": "75px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          >
            <div
              className="rt-td col-user"
              role="gridcell"
              style={
                Object {
                  "flex": "50 0 auto",
                  "maxWidth": null,
                  "width": "50px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-action"
              role="gridcell"
              style={
                Object {
                  "flex": "50 0 auto",
                  "maxWidth": null,
                  "width": "50px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-id"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": "100px",
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-initiator"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-respondent"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-timestamp"
              role="gridcell"
              style={
                Object {
                  "flex": "75 0 auto",
                  "maxWidth": null,
                  "width": "75px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          >
            <div
              className="rt-td col-user"
              role="gridcell"
              style={
                Object {
                  "flex": "50 0 auto",
                  "maxWidth": null,
                  "width": "50px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-action"
              role="gridcell"
              style={
                Object {
                  "flex": "50 0 auto",
                  "maxWidth": null,
                  "width": "50px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-id"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": "100px",
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-initiator"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-respondent"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-timestamp"
              role="gridcell"
              style={
                Object {
                  "flex": "75 0 auto",
                  "maxWidth": null,
                  "width": "75px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          >
            <div
              className="rt-td col-user"
              role="gridcell"
              style={
                Object {
                  "flex": "50 0 auto",
                  "maxWidth": null,
                  "width": "50px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-action"
              role="gridcell"
              style={
                Object {
                  "flex": "50 0 auto",
                  "maxWidth": null,
                  "width": "50px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-id"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": "100px",
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-initiator"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-respondent"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-timestamp"
              role="gridcell"
              style={
                Object {
                  "flex": "75 0 auto",
                  "maxWidth": null,
                  "width": "75px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          >
            <div
              className="rt-td col-user"
              role="gridcell"
              style={
                Object {
                  "flex": "50 0 auto",
                  "maxWidth": null,
                  "width": "50px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-action"
              role="gridcell"
              style={
                Object {
                  "flex": "50 0 auto",
                  "maxWidth": null,
                  "width": "50px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-id"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": "100px",
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-initiator"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-respondent"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td col-timestamp"
              role="gridcell"
              style={
                Object {
                  "flex": "75 0 auto",
                  "maxWidth": null,
                  "width": "75px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="pagination-bottom"
    >
      <div
        className="-pagination"
      >
        <div
          className="-previous"
        >
          <button
            className="-btn"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            Previous
          </button>
        </div>
        <div
          className="-center"
        >
          <span
            className="-pageInfo"
          >
            Page
             
            <div
              className="-pageJump"
            >
              <input
                onBlur={[Function]}
                onChange={[Function]}
                onKeyPress={[Function]}
                type="number"
                value={1}
              />
            </div>
             
            of
             
            <span
              className="-totalPages"
            >
              1
            </span>
          </span>
          <span
            className="select-wrap -pageSizeOptions"
          >
            <select
              onChange={[Function]}
              value={10}
            >
              <option
                value={5}
              >
                5
                 
                rows
              </option>
              <option
                value={10}
              >
                10
                 
                rows
              </option>
              <option
                value={15}
              >
                15
                 
                rows
              </option>
              <option
                value={20}
              >
                20
                 
                rows
              </option>
              <option
                value={25}
              >
                25
                 
                rows
              </option>
              <option
                value={50}
              >
                50
                 
                rows
              </option>
              <option
                value={100}
              >
                100
                 
                rows
              </option>
            </select>
          </span>
        </div>
        <div
          className="-next"
        >
          <button
            className="-btn"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            Next
          </button>
        </div>
      </div>
    </div>
    <div
      className="rt-noData"
    >
      No rows found
    </div>
    <div
      className="-loading -active"
    >
      <div
        className="-loading-inner"
      >
        Loading...
      </div>
    </div>
  </div>
</div>
`;
